### DISCUSSION TITLE
Add existing PDF as background

### DISCUSSION THREAD

**Discussion by joost-leisink** [QuestPDF/QuestPDF#843] [Q&A] [answered]
Is it possible to add an existing PDF as background for a new PDF that I create?
If so, how would I need to do this?

**Comment by girlpunk** [comment]
Unfortunately, QuestPDF does not provide any support for editing or opening existing PDFs. You may be able to use another library to render the existing PDF into an image (either raster or vector) and use that as a background in QuestPDF.

**Comment by MarcinZiabek** [comment, marked as answer]
I am currently preparing a long-term roadmap for the library. I am excited to include this enhancement 😄

Most likely, the work will be done by using already existing C++ libraries and surfacing their API to the managed C# code. Depending on the complexity and value-added, it will be done as part of QuestPDF or a secondary MIT library.

**Reply by kavatari** [reply]
Would love to see this feature in QuestPDF.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

This functionality has been available since version 2024.12.0. You can use an existing PDF as a background by underlaying it beneath your generated content with `UnderlayFile`, and the `RepeatSourcePages` option lets a single background page repeat across every page. You can find the details in the [Document Operations](https://www.questpdf.com/concepts/document-operations.html) documentation.

Should you encounter any further problems, please don't hesitate to open a new issue.
Thank you!
